src/query/src/dist_plan/merge_scan.rs

```diff
@@ -150,40 +150,36 @@ impl MergeScanExec {
         let metric = MergeScanMetric::new(&self.metric);
 
         let stream = try_stream! {
+            let _finish_timer = metric.finish_time().timer();
             for peer in peers {
                 let client = clients.get_client(&peer).await;
                 let database = Database::new(&table.catalog_name, &table.schema_name, client);
-                let _timer = metric.grpc_time().timer();
+
+                let mut ready_timer = metric.ready_time().timer();
+                let mut first_consume_timer = metric.first_consume_time().timer();
```

Comment:
I think we should move `ready_timer` and `first_consume_timer` out of for loop, too. If there are multiple peers, we must make sure the timers are recorded exactly once.